src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -69,14 +71,29 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private validateName(name: string): string | null | undefined {
+    private async validateStoredProcedureName(name: string, getChildrenTask: Promise<AzExtTreeItem[]>): Promise<string | null | undefined> {
+        const currStoredProcedureList = await getChildrenTask;
+        const currStoredProcedureNames: string[] = [];
+        for (const sp of currStoredProcedureList) {
+            if (sp instanceof DocDBStoredProcedureTreeItem) {
+                currStoredProcedureNames.push(sp.id);
+            }
+        }
         if (name) {
-            if (name.indexOf("/") !== -1 || name.indexOf("\\") !== -1 || name.indexOf("?") !== -1 || name.indexOf("#") !== -1) {
-                return "Id contains illegal chars: /,\\,?,#";
+            if (name.length < 1 || name.length > 255) {
+                return "Name has to be between 1 and 255 chars long";
+            }
+            if (/[/\\?#&]/.test(name)) {
+                return localize("illegalChars", "Name contains illegal chars: /,\\,?,#,&");
             }
             if (name[name.length - 1] === " ") {
-                return "Id ends with a space.";
+                return localize("endsWithSpace", "Name ends with a space.");
```

Comment:
nit: I think it makes more sense to add a "cannot" here
```suggestion
                return localize("endsWithSpace", "Name cannot end with a space.");
```